scripts/accounts-invariants/pallet-balances-maintenance.py

```diff
@@ -333,6 +356,245 @@ def upgrade_accounts(chain_connection,
 
 
+def check_if_account_has_double_providers(account, chain_major_version, ed):
+    """
+    This predicate checks if an account's providers counter is equal to 2
+    :param account: AccountInfo struct (element of System.Accounts StorageMap)
+    :param chain_major_version: Must be >= 12
+    :param ed: existential deposit, not used
+    :return: True if providers == 2, False otherwise
+    """
+
+    assert chain_major_version == ChainMajorVersion.AT_LEAST_12_MAJOR_VERSION, \
+        "Chain major version must be at least 12!"
+
+    providers = account['providers'].value
+
+    return providers == 2
+
+
+def state_sanity_check(chain_connection,
+                       account_id,
+                       set_storage_block_hash):
+    """
+    Makes sure AccountInfo data changed only in terms of providers counter by comparing state between block that
+    setStorage landed vs state of that block parent.
+    :param chain_connection WS connection handler
+    :param account_id: An account to check
+    :param set_storage_block_hash: a hash of a block that contains setStorage
+    :return: None.
+    """
+    set_storage_block_number = chain_connection.get_block_number(set_storage_block_hash)
+    parent_block_number = set_storage_block_number - 1
+    parent_block_hash = chain_connection.get_block_hash(parent_block_number)
+
+    log.info(f"Comparing AccountInfo for account {account_id} "
+             f"between blocks #{parent_block_number} and #{set_storage_block_number}")
+    parent_account_info_state = chain_connection.query(module='System',
+                                                       storage_function='Account',
+                                                       params=[account_id],
+                                                       block_hash=parent_block_hash).value
+    log.debug(f"Account state in the parent block: {parent_account_info_state}")
+    set_storage_info_state = chain_connection.query(module='System',
+                                                    storage_function='Account',
+                                                    params=[account_id],
+                                                    block_hash=set_storage_block_hash).value
+    log.debug(f"Account state in the setStorage block: {set_storage_info_state}")
+    assert parent_account_info_state['providers'] == 2, f"Providers before fix for account {account_id} is not 2!"
+    assert set_storage_info_state['providers'] == 1, f"Providers after fix for account {account_id} is not 1!"
+    changed_state_for_sake_of_assert = copy.deepcopy(parent_account_info_state)
+    changed_state_for_sake_of_assert['providers'] = 1
+    assert changed_state_for_sake_of_assert == set_storage_info_state, \
+        f"Parent account info state is different from set storage state with more than providers counter! " \
+        f"Parent state: {parent_account_info_state} " \
+        f"Set storage state: {set_storage_info_state}"
```

Comment:
You can also print `set_storage_block_number` so that it's easier to find, in case it fires

Reply:
done